packages/cli/test/unit/services/cache.service.test.ts

```diff
@@ -0,0 +1,125 @@
+import Container from 'typedi';
+import { CacheService } from '@/services/cache.service';
+import type { MemoryCache } from 'cache-manager';
+// import type { RedisCache } from 'cache-manager-ioredis-yet';
+import config from '@/config';
+
+const cacheService = Container.get(CacheService);
+
+describe('cacheService', () => {
+	beforeEach(async () => {
+		await Container.get(CacheService).uninit();
+	});
+
+	test('should create a memory cache by default', async () => {
+		config.set('executions.mode', 'regular');
+		config.set('cache.backend', 'auto');
+		await cacheService.init();
+		expect(cacheService.getCacheInstance()).toBeDefined();
+		const candidate = cacheService.getCacheInstance() as MemoryCache;
```

Comment:
We can also use `fail()` to narrow down the type, since we know it won't be `undefined` by now.